services/src/main/java/org/fao/geonet/api/records/attachments/AttachmentsApi.java

```diff
@@ -194,6 +195,10 @@ public MetadataResource putResource(
         @Parameter(description = "The file to upload") @RequestParam("file") MultipartFile file,
         @Parameter(description = "Use approved version or not", example = "true") @RequestParam(required = false, defaultValue = "false") Boolean approved,
         @Parameter(hidden = true) HttpServletRequest request) throws Exception {
+        if (org.apache.commons.lang3.StringUtils.contains(file.getOriginalFilename(),';')) {
```

Comment:
Maybe this check could be done in AbstractStore#checkResourceId (or FilePathChecker#verify) ? Probably in checkResourceId

Reply:
Probably a good idea to put business logic into the implementation layer. I will refactoring the code a little.